src/main/java/com/contrastsecurity/sdk/scan/CodeArtifacts.java

```diff
@@ -62,6 +62,7 @@
    */
   CodeArtifact upload(Path file, String name) throws IOException;
 
+  CodeArtifact upload(Path file, String name, Path metadata, String metaname) throws IOException;
```

Comment:
What is `metaname` here? I don't think it's used.

Reply:
`metaname is analagous to the `name` argument the way `metadata` is analogous to the `file` argument.  As to if that is even needed, I'm not sure.  I don't see why we'd give the user the capability to change the "filename" on the multipart upload... but I figured it was something I didn't understand so I just mirrored the behavior when allowing the prescan metadata to be added to the multipart file upload.  It's your call if you want me to take it out as I don't have a valid argument for or against it.

Reply:
ok, I see now.  I can take it out.  Before I do any work on it yet though I'll wait until we resolve the fundamental API design questions you have in this area.